pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -144,13 +153,47 @@ func (e *CloudWatchExecutor) executeTimeSeriesQuery(ctx context.Context, queryCo
 
 	if len(getMetricDataQueries) > 0 {
 		for region, getMetricDataQuery := range getMetricDataQueries {
+			// restore referenced query for alerting
+			parentQuery, err := parseQuery(queryContext.Queries[0].Model, defaultRegion)
+			RefId := queryContext.Queries[0].RefId
+			if err != nil {
+				results.Results[RefId] = &tsdb.QueryResult{
+					Error: err,
+				}
+				return results, nil
+			}
+
+			alerting := false
+			if len(getMetricDataQueries) == 1 && len(getMetricDataQuery) == 1 {
+				alerting = true
```

Comment:
is there away to avoid doing this? I think if there is a targetFull use that. Don't like the special alerting checks here.

Reply:
to get MetricMath result, I need to send all related query to CloudWatch endpoint.
Then, the endpoint response multiple correspond time series.
Grafana doesn't allow to include those correspond time series in alert response, so I need to filter it by `alerting` flag.

Reply:
That does not explain what I am asking here, can't you always use targetFull (in alerting and normal dashboard queries) ? Then you do not need a special handling case

Reply:
can't use `targetFull` if query contain template variable.
If CloudWatch frontend code expand template variable to `targetFull`, it may work, but the code becomes more complicated...

Reply:
not sure I understand, does the cloudwatch backend interpolate variables? Is the variables sent to the server, how?

Reply:
I should explain about the problem in more detail, sorry.

CloudWatch frontend code interpolate variables. I refer the PR, and the line where interpolation occured.
https://github.com/grafana/grafana/pull/5003/files#diff-a8259c8cbce476f0faab44ef81675dfbR26

The interpolation result is not saved to targetFull because frontend code interpolate variables when sending query to backend.

In this PR, targetFull is updated when user edit panel.
To remove `special alerting checks`, the frontend code need to render targetFull after interpolate variables.
The interpolation code is already complicated, and it had several bug in past version.

The rendering targetFull code is also complicated, mixing these code is not good idea, I think.

Reply:
ok, thanks for clarifying this @mtanda . think I understand. But not really happy with how this fix looks. Feels very hacky & fragile (not super happy with how graphite solves this either), but this slightly worse. Might need to find a more robust way to refer to other queries in a alert query.